Sources/SpeziHealthKit/CollectSample/HealthKitSampleDataSource.swift

```diff
@@ -111,35 +107,34 @@ final class HealthKitSampleDataSource: HealthKitDataSource {
             case .background:
                 active = true
                 try await healthStore.startBackgroundDelivery(for: [sampleType]) { result in
-                    Task {
```

Comment:
I am not quite sure if it was by-design to add a task here to ensure that it is not blocking the `triggerManualDataSourceCollection` function to eventually return. But as far as I can see this it never could and should to the same as the other branches.

Reply:
I think the Task was due to the async call to `try await self.anchoredSingleObjectQuery()` at the bottom of the closure.
The Task is still created, however we create as bart of the `startBackgroundDelivery` method, as otherwise the isolation checker would think that we are sending `self` which is not Sendable.

Reply:
So functionally is not different. It's just a bit differently structured to provide better/clearer isolation guarantees (And to workaround HealthKit's missing concurrency annotations).